src/goto-programs/goto_program.h

```diff
@@ -539,8 +539,11 @@ class goto_programt
       return nullopt;
     }
 
-    return std::next(
+    auto location_target = std::next(
       instructions.begin(), location_number - start_location_number);
+
+    CHECK_RETURN(location_target->location_number == location_number);
```

Comment:
You added this based on this comment: "I'm not sure we provide a guarantee that location numbers are contiguous. You might want to add a CHECK_RETURN to make sure what you are about to return has the expected location_number."

possible comment:
"location numbers may not be contiguous. Verify expected location number holds:

If who wrote this did not know it -  then who reviews it or uses it in the future may also not. So clarify why you added this or they will be looking at it going - "why?"

Take these as opportunities to provide clarify - not just fixes. 

*Pass on what you know or have found out... *

Reply:
Added.